### DISCUSSION TITLE
URI list via core API

### DISCUSSION THREAD

**Discussion by marioaag** [avocado-framework/avocado#4565] [Q&A] [answered]
Is there a simple way to get the list of references via python core API when a test_file or test_file:class is provided?

**Comment by beraldoleal** [comment]
Hi @marioaag, for each test suite you can query the `references` object:

```python3
suite.references
```

Or via the config dict:

```python3
suite.config.get('run.references')
```

Is that what you are looking for? If not, could you provide a snippet of what are you doing so I can understand better?

**Reply by marioaag** [reply]
Hi @beraldoleal, I'm looking for something before the creation of the suite. A function that I can use to provide a reference and get all the URIs for it like doing:

```
avocado list /tests_dir/test_file.py
```

And get a list of the URIs for all the instrumented tests

Looking for this due in the creations of the suite I'm facing that it doesn't support reference specification it needs the full URI

**Reply by beraldoleal** [reply, marked as answer]
Ohh, maybe you are looking for something like this:

```python
from avocado.core.job import Job

config = {'run.references': ['selftests/unit/test_suite.py']}

with Job.from_config(config) as job:
   print(job.test_suite.tests)
```

This will give you this output:

```python
[(avocado.core.test.PythonUnittest,
  {'name': 'test_suite.TestSuiteTest.test_custom_suite',
   'test_dir': '/home/local/src/beraldo/avocado/selftests/unit',
   'tags': {}}),
 (avocado.core.test.PythonUnittest,
  {'name': 'test_suite.TestSuiteTest.test_automatic_suite',
   'test_dir': '/home/local/src/beraldo/avocado/selftests/unit',
   'tags': {}}),
 (avocado.core.test.PythonUnittest,
  {'name': 'test_suite.TestSuiteTest.test_config_extend_manual',
   'test_dir': '/home/local/src/beraldo/avocado/selftests/unit',
   'tags': {}}),
 (avocado.core.test.PythonUnittest,
  {'name': 'test_suite.TestSuiteTest.test_config_extend_automatic',
   'test_dir': '/home/local/src/beraldo/avocado/selftests/unit',
   'tags': {}})]
```

Is that handy?

**Reply by marioaag** [reply]
Thanks, that helps! but I was expecting something that returns the URI as is and not need to post process this info, for sure this logic is already somewhere due avocado list print it and also when TestSuite.from_config generate them URIs itself.

I want to pre-generate the URIs to generate my own Runnable due the Runnable class doesn't have the ability to generate them base in a test reference
